app/models/project/storage.rb

```diff
@@ -0,0 +1,109 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+module Project::Storage
+  def self.included(base)
+    base.send :extend, StorageMethods
+    base.send :include, ModelMethods
+  end
+
+  module ModelMethods
+    ##
+    # Count required disk storage for this project.
+    # Returns a hash of the form:
+    # 'total' => Total required disk space for this project
+    # 'modules' => Hash of localized keys and required space for this module
+    def count_required_storage
+      storage = self.class.with_required_storage.find(id)
+
+      {
+        'total' => storage.required_disk_space.to_i,
+        'modules' => {
+          'label_work_package_plural' => storage.work_package_required_space.to_i,
+          'project_module_wiki' => storage.wiki_required_space.to_i,
+          'label_repository' => storage.repositories_required_space.to_i
+        }.select{ |k,v| v > 0 }
```

Comment:
Unused block argument - `k`. If it's necessary, use `_` or `_k` as an argument name to indicate that it won't be used.<br>Space missing after comma.<br>Space missing to the left of {.